### PR TITLE
revenue sharing POC

### PR DESCRIPTION
**Pull request by mortelli**
PR to merge work between myself and @antomor for the revenue sharing POC.

### CODE REVIEW

**Review by mortelli — COMMENTED** [review]
LGTM

File: src/sdk.ts [hunk 1]
```diff
@@ -237,7 +237,7 @@ export class DefaultRelayingServices implements RelayingServices {
             tokenAddress,
             tokenAmount
         });
-        tokenAmount = tokenAmount ?? 0;
+        // tokenAmount = tokenAmount ?? 0;
```
Comment (mortelli) [inline src/sdk.ts:240]: this param is optional, right? are we sure we want to remove this assignment? 

if `undefined`, won't line `277` cause an error?

Comment (antomor) [reply src/sdk.ts:240]: No, it won't. The following statement `a?.toString()` is similar to `a ?: a.toString():a` with some small differences

Comment (mortelli) [reply src/sdk.ts:240]: my bad, tried it on the javascript console and it gave me an error, i suppose it's a ts thing :stuck_out_tongue:

**Pull request merged**